### PR TITLE
drop everything but {End,}Key from roachpb.RequestHeader

### PR DESCRIPTION
**Pull request by tbg**
another major refactor. It won't be fun to review, but at least the individual commits are reasonably self-contained.
I'm going to give this a close review myself tomorrow (there are probably some typos/...), but someone else definitely should do the same. Mild concern: Changes in the tests may have changed their semantics (dropping a Txn somewhere, sending to the wrong Range) - I've paid close attention, but with the hundreds of tests touched here something might've gone wrong.

### CODE REVIEW

File: storage/client_test.go [hunk 1]
```diff
@@ -51,6 +51,15 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+func rg1(s *storage.Store) client.Sender {
+	return client.Wrap(s, func(ba roachpb.BatchRequest) roachpb.BatchRequest {
+		if ba.RangeID == 0 {
+			ba.RangeID = 1
```
Comment (vivekmenezes) [inline storage/client_test.go]: why did you need this?

Comment (tschottdorf) [inline storage/client_test.go]: Convenience. Most tests send to that range and it made the refactor easier, plus saving a lot of LOC.

Comment (bdarnell) [inline storage/client_test.go]: I've wanted to do something like this for a long time, although I was planning to look up the range ID based on the keys in the request (kind of like a mini DistSender). We could even put the lookup on the Store itself so we wouldn't have to use a test-specific wrapper.

**Comment by bdarnell** [comment]
Nit: `Store.testSender` is introduced in the commit "refactor storage tests", not the commit "Introduce Store.testSender"

**Comment by bdarnell** [comment]
LGTM

**Comment by vivekmenezes** [comment]
Nice change!

File: storage/client_test.go (hunk 1 above)
Comment (tschottdorf) [inline storage/client_test.go]: I did that in the internal tests, but this is `storage_test`. I had intermittently introduced `(*Store).TestSender()` but didn't want to leak obvious test code into the public interface. Agreed that this looks clumsier than it should be - how would you go about it?

**Comment by tschottdorf** [comment]
@bdarnell reworded the commit.

File: storage/client_test.go (hunk 1 above)
Comment (bdarnell) [inline storage/client_test.go]: By "on the Store itself" I meant `Store.Send`, not a new method like `Store.TestSender`. It would put code intended for tests in the production code path, but we already do something similar in LocalSender to look up missing store IDs.
